### ISSUE TITLE
Can't use validation_data when training with single time series? (DeepAR)

### ISSUE THREAD

**Issue by mariosantosprivate** [awslabs/gluon-ts#1158]
Hello, so I have several time series of daily sales spanning 940 days, the goal being to predict the last 40 days, and I've split it into 3 sets:

- ListDataSet train_ds : First 860 days
- ListDataSet val_ds: First 900 days
- ListDataSet test_ds: All records (940)

I've trained a Deep AR model using `validation_data = val_ds` and it worked fine for multiple time series, however when I only train it on a single time series, after a couple of epochs I get the following error:

```python
  0%|          | 0/50 [00:00<?, ?it/s]learning rate from ``lr_scheduler`` has been overwritten by ``learning_rate`` in optimizer.
100%|██████████| 50/50 [02:00<00:00,  2.42s/it, epoch=1/100, avg_epoch_loss=8.9] 
1it [00:00,  2.50it/s, epoch=1/100, validation_avg_epoch_loss=8.6]
100%|██████████| 50/50 [00:11<00:00,  4.27it/s, epoch=2/100, avg_epoch_loss=8.68]
1it [00:00,  2.67it/s, epoch=2/100, validation_avg_epoch_loss=8.62]
100%|██████████| 50/50 [00:11<00:00,  4.38it/s, epoch=3/100, avg_epoch_loss=8.66]
1it [00:00,  2.72it/s, epoch=3/100, validation_avg_epoch_loss=8.59]
100%|██████████| 50/50 [00:11<00:00,  4.35it/s, epoch=4/100, avg_epoch_loss=8.65]
1it [00:00,  2.71it/s, epoch=4/100, validation_avg_epoch_loss=8.42]
100%|██████████| 50/50 [00:11<00:00,  4.41it/s, epoch=5/100, avg_epoch_loss=8.61]
0it [00:00, ?it/s]
---------------------------------------------------------------------------
UnboundLocalError                         Traceback (most recent call last)
<ipython-input-36-8857ef7fe6cd> in <module>()
     21 # Using a validation set training stops as soon as the loss on the validation set (and therefore on unseen data) does not decrease any further.
     22 deepAR_predictor = deepAR_estimator.train(training_data=train_ds, 
---> 23                                           validation_data=val_ds
     24                                           )
     25 filepath = '/content/gdrive/My Drive/Thesis/Models/deepAR_single_ts.pkl'

3 frames
/usr/local/lib/python3.6/dist-packages/gluonts/mx/trainer/_base.py in loop(epoch_no, batch_iter, is_training)
    324                         epoch_no,
    325                         ("" if is_training else "validation_") + "epoch_loss",
--> 326                         lv,
    327                     )
    328 

UnboundLocalError: local variable 'lv' referenced before assignment
```
If I don't use the `validation_data`, and just give the model the `val_ds `as the training set it works fine.
```python
deepAR_predictor = deepAR_estimator.train(training_data=val_ds)
```

Am I doing something wrong?

**Comment by kaijennissen** [comment]
@mariosantosprivate See #997 for more info.

**Comment by lostella** [comment]
fixed in #1256
